src/azure-cli/azure/cli/command_modules/monitor/operations/monitor_clone_util.py

```diff
@@ -2,98 +2,120 @@
+# pylint: disable=line-too-long, protected-access
 from msrestazure.tools import parse_resource_id
 from knack.log import get_logger
 from knack.util import CLIError
-
-from .._client_factory import cf_monitor
-from ..util import gen_guid
+from azure.cli.core.commands.transform import _parse_id
+from azure.cli.command_modules.network.custom import _convert_to_snake_case
+from azure.cli.command_modules.monitor.util import gen_guid
 
 logger = get_logger(__name__)
 CLONED_NAME = "cloned-{}-{}"
 
 
-def _gen_metrics_alert_rules_clone_list(monitor_client, source_resource, target_resource):
-    alert_rules = list(monitor_client.metric_alerts.list_by_subscription())
+def _get_metrics_alert_rules_clone_list(cmd, source_resource, target_resource):
+    subscription_id = parse_resource_id(source_resource)['subscription']
+    from ..aaz.latest.monitor.metrics.alert import List
+    alert_rules = List(cli_ctx=cmd.cli_ctx)(command_args={"subscription": subscription_id})
+    alert_rules = _convert_to_snake_case(alert_rules)
     for alert_rule in alert_rules:
-        if source_resource in alert_rule.scopes:
-            if target_resource not in alert_rule.scopes:
+        if source_resource in alert_rule['scopes']:
+            if target_resource not in alert_rule['scopes']:
                 yield alert_rule
             else:
                 logger.warning('The target resource already has alert rule %s. '
-                               'Skip cloning this one.', alert_rule.name)
-
-
-def _add_into_existing_scopes(monitor_client, alert_rule, target_resource):
-    alert_rule.scopes.append(target_resource)
-    resource_group_name, name = _parse_id(alert_rule.id).values()  # pylint: disable=unbalanced-dict-unpacking
-    return monitor_client.metric_alerts.create_or_update(resource_group_name=resource_group_name,
-                                                         rule_name=name,
-                                                         parameters=alert_rule)
-
-
-def _clone_and_replace_action_group(source_monitor_client, target_monitor_client,
-                                    cmd, alert_rule, action_group_mapping, target_resource):
-    for index, action in enumerate(alert_rule.actions):
-        if action.action_group_id in action_group_mapping:
-            alert_rule.actions[index] = action_group_mapping[action.action_group_id][1]
+                               'Skip cloning this one.', alert_rule['name'])
+
+
+def _add_into_existing_scopes(cmd, source_resource, alert_rule, target_resource):
+    subscription_id = parse_resource_id(source_resource)['subscription']
+    alert_rule['scopes'].append(target_resource)
+    resource_group_name, name = _parse_id(alert_rule['id']).values()  # pylint: disable=unbalanced-dict-unpacking
+    alert_rule["subscription"] = subscription_id
+    alert_rule["resource_group"] = resource_group_name
+    alert_rule["name"] = name
+    from ..aaz.latest.monitor.metrics.alert import Update
```

Comment:
Is there need `format_metrics_alert_req(alert_rule)` as well?

Reply:
Only `scopes` are updated so no need to format other parameters, they are useless in `Update` cmd

Reply:
In that case, please specify the arg names passed into command_args. Such as 
```
command_args={
  "name": name,
  "resource_group": resource_group,
  "subscription": subscription_id,
  "scopes": scopes,
}
```